nilearn/surface.py

```diff
@@ -0,0 +1,572 @@
+       an image (represented as a column vector) gives the projection onto mesh
+       vertices.
+
+    """
+    if mask is not None and tuple(mask.shape) != img_shape:
+        raise ValueError('mask should have shape img_shape')
+    mesh = load_surf_mesh(mesh)
+    sample_locations = _sample_locations(
+        mesh, affine, kind=kind, radius=radius, n_points=n_points)
+    sample_locations = np.asarray(np.round(sample_locations), dtype=int)
+    n_vertices, n_points, img_dim = sample_locations.shape
+    masked = _masked_indices(np.vstack(sample_locations), img_shape, mask=mask)
+    sample_locations = np.rollaxis(sample_locations, -1)
+    sample_indices = np.ravel_multi_index(
+        sample_locations, img_shape, mode='clip').ravel()
+    row_indices, _ = np.mgrid[:n_vertices, :n_points]
+    row_indices = row_indices.ravel()
+    row_indices = row_indices[~masked]
+    sample_indices = sample_indices[~masked]
+    weights = np.ones(len(row_indices))
+    proj = sparse.csr_matrix(
+        (weights, (row_indices, sample_indices.ravel())),
+        shape=(n_vertices, np.prod(img_shape)))
+    proj = sklearn.preprocessing.normalize(proj, axis=1, norm='l1')
+    return proj
+
+
+def _nearest_voxel_sampling(images, mesh, affine, kind='ball', radius=3.,
+                            n_points=None, mask=None):
+    """In each image, measure the intensity at each node of the mesh.
+
+    Image intensity at each sample point is that of the nearest voxel.
+    A 2-d array is returned, where each row corresponds to an image and each
+    column to a mesh vertex.
+    See documentation of vol_to_surf for details.
+
+    """
+    proj = projection_matrix(
+        mesh, affine, images[0].shape, kind=kind, radius=radius,
+        n_points=n_points, mask=mask)
+    data = np.asarray(images).reshape(len(images), -1).T
+    texture = proj.dot(data)
+    texture[np.asarray(proj.sum(axis=1) == 0).ravel()] = np.nan
+    return texture.T
+
+
+def _interpolation_sampling(images, mesh, affine, kind='ball', radius=3,
+                            n_points=None, mask=None):
+    """In each image, measure the intensity at each node of the mesh.
+
+    Image intensity at each sample point is computed with trilinear
+    interpolation.
+    A 2-d array is returned, where each row corresponds to an image and each
+    column to a mesh vertex.
+    See documentation of vol_to_surf for details.
+
+    """
+    sample_locations = _sample_locations(
+        mesh, affine, kind=kind, radius=radius, n_points=n_points)
+    n_vertices, n_points, img_dim = sample_locations.shape
+    grid = [np.arange(size) for size in images[0].shape]
+    interp_locations = np.vstack(sample_locations)
+    masked = _masked_indices(interp_locations, images[0].shape, mask=mask)
+    # loop over images rather than building a big array to use less memory
+    all_samples = []
+    for img in images:
+        interpolator = interpolate.RegularGridInterpolator(
```

Comment:
So, this will fall back to nearest interpolation when scipy is < 0.14 and linear is given. Isn't it ?

Reply:
yes, this function will never be called if scipy < 0.14